solara/website/pages/documentation/components/viz/plotly.py

```diff
@@ -18,46 +18,54 @@
 
 @solara.component
 def Page():
-    with solara.VBox() as main:
-        selection_data, set_selection_data = solara.use_state(None)
-        click_data, set_click_data = solara.use_state(None)
-        hover_data, set_hover_data = solara.use_state(None)
-        unhover_data, set_unhover_data = solara.use_state(None)
-        deselect_data, set_deselect_data = solara.use_state(None)
-        fig = px.scatter(df, x="sepal_width", y="sepal_length", color="species")
-        solara.FigurePlotly(
-            fig, on_selection=set_selection_data, on_click=set_click_data, on_hover=set_hover_data, on_unhover=set_unhover_data, on_deselect=set_deselect_data
-        )
-
-        solara.Markdown(
-            f"""
+    state = solara.use_reactive(
+        {
+            "selection_data": None,
+            "click_data": None,
+            "hover_data": None,
+            "unhover_data": None,
+            "deselect_data": None,
+        }
+    )
+
+    fig = px.scatter(df, x="sepal_width", y="sepal_length", color="species")
+    solara.FigurePlotly(
+        fig,
+        on_selection=lambda data: state.value.update({"selection_data": data}),
+        on_click=lambda data: state.value.update({"click_data": data}),
+        on_hover=lambda data: state.value.update({"hover_data": data}),
+        on_unhover=lambda data: state.value.update({"unhover_data": data}),
+        on_deselect=lambda data: state.value.update({"deselect_data": data}),
```

Comment:
The `.update` method on a dictionary mutates that dictionary, which means that this won't result in the state being updated. Instead, we should explicitly `.set` the state with the previous values and the new value, i.e.

```suggestion
        on_selection=lambda data: state.set({**state.value, "selection_data": data}),
        on_click=lambda data: state.set({**state.value, "click_data": data}),
        on_hover=lambda data: state.set({**state.value, "hover_data": data}),
        on_unhover=lambda data: state.set({**state.value, "unhover_data": data}),
        on_deselect=lambda data: state.set({**state.value, "deselect_data": data}),
```